src/app/component/market/market-position-detail.tsx

```diff
@@ -0,0 +1,176 @@
+"use client";
+import Image from "next/image";
+import { useRouter } from "next/navigation";
+import etheriumIcon from "../../../../public/icons/etherium.svg";
+import { positionCardsData } from "@/data";
+import useElementWidth from "@/hooks/useElementWidth";
+import Card from "../common/Card";
+import { useEffect, useState } from "react";
+import MarketModal from "./marketModal";
+import Slider from "react-slick";
+
+export const MarketPositionDetail = () => {
+  const router = useRouter();
+  const [isPositionName, setIsPositionName] = useState(false);
+  const [sectionWidth, sectionRef] = useElementWidth<HTMLDivElement>();
+  const [slidesToShow, setSlidesToShow] = useState<number>(3);
+
+  useEffect(() => {
+    if (sectionWidth) {
+      if (sectionWidth > 1800) {
+        setSlidesToShow(5);
+      } else if (sectionWidth < 768 && sectionWidth > 350) {
+        setSlidesToShow(2);
+      } else if (sectionWidth > 1200 && sectionWidth <= 1800) {
+        setSlidesToShow(4);
+      } else if (sectionWidth > 768 && sectionWidth <= 1200) {
+        setSlidesToShow(3);
+      } else {
+        setSlidesToShow(1);
+      }
```

Comment:
This could be a helper that you pass an array and it calculates the right value to be picked.

Reply:
done